diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ExpoCropImageUtils.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ExpoCropImageUtils.kt
index a876bbc..3bb7e10 100644
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ExpoCropImageUtils.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ExpoCropImageUtils.kt
@@ -81,7 +81,9 @@ object ExpoCropImageUtils {
     toolbarColor: Int,
     isNight: Boolean
   ) {
-    window.statusBarColor = toolbarColor
+    if (android.os.Build.VERSION.SDK_INT < 35) {
+      window.statusBarColor = toolbarColor
+    }
     WindowInsetsControllerCompat(window, window.decorView).isAppearanceLightStatusBars = !isNight
   }
 }
